sqlserver/datadog_checks/sqlserver/deadlocks.py

```diff
@@ -104,13 +114,39 @@ def _get_lookback_seconds(self):
     def _get_connector(self):
         return self._check.connection.connector
 
+    def _set_xe_session_name(self):
+        with self._check.connection.open_managed_default_connection(key_prefix=self._conn_key_prefix):
+            with self._check.connection.get_managed_cursor(key_prefix=self._conn_key_prefix) as cursor:
+                if self._xe_session_name is None:
+                    cursor.execute(XE_SESSIONS_QUERY)
+                    rows = cursor.fetchall()
+                    if not rows:
+                        raise Exception(NO_XE_SESSION_ERROR)
```

Comment:
I suggest to create a custom exception `class NoXeSessionException(Exception)` so that we can catch the specific exception when needed

Reply:
btw i also don't see error handling of in `_set_xe_session_name` and `_query_deadlocks`. if an exception is raised here, it will fail the thread. should we catch the exception and log an error message?

Reply:
The custom exception NoXeSessionException is now raised in _set_xe_session_name and caught in _query_deadlocks.